src/main/java/baritone/command/defaults/ToMapCommand.java

```diff
@@ -0,0 +1,223 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.command.defaults;
+
+import baritone.api.IBaritone;
+import baritone.api.cache.ICachedRegion;
+import baritone.api.command.Command;
+import baritone.api.command.argument.IArgConsumer;
+import baritone.api.command.exception.CommandException;
+import net.minecraft.block.AirBlock;
+import net.minecraft.block.BlockState;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class ToMapCommand extends Command {
+
+    private HashMap<String, Color> cachedColors;
+
+    public ToMapCommand(IBaritone baritone, String... names) {
+        super(baritone, "tomap");
+        cachedColors = new HashMap<>();
+    }
+
+    @Override
+    public void execute(String label, IArgConsumer args) throws CommandException {
+        ctx.worldData().getCachedWorld().reloadAllFromDisk();
+        ICachedRegion cr = ctx.worldData().getCachedWorld().getRegion(0, 0);
+        Map<String, int[]> cornerBlockCoordinatesForZeroZero = getBlockCoordsOfRegionCorners(0, 0);
+        int[] nw = cornerBlockCoordinatesForZeroZero.get("NW");
+        Map<Integer, Map<Integer, String>> topView = new LinkedHashMap<>();
+        Pattern p = Pattern.compile("Block\\{(.*)\\}");
+        for (int x = nw[0]; x < nw[0] + 512; x++) {
+            if (!topView.containsKey(x)) {
+                HashMap<Integer, String> zId = new LinkedHashMap<>();
+                topView.put(x, zId);
+            }
+            for (int z = nw[1]; z < nw[1] + 512; z++) {
+                for (int y = 256; y > 0; y--) {
+                    BlockState bs = cr.getBlock(x, y, z);
+                    String id;
+                    if (bs == null) {
+                        id = "air";
+                    } else {
+                        try {
+                            Matcher m = p.matcher(bs.toString());
+                            if (m.find())
+                                id = m.group(1);
+                            else
+                                throw new Exception("Damn");
```

Comment:
Maybe give more information on what caused the exception

Reply:
Agreed